.evergreen/scripts/compile-scan-build.sh

```diff
@@ -67,14 +67,18 @@ scan_build_flags=(
   --exclude "$(pwd)/build/_deps"                         # mongoc
 )
 
-case "${BSONCXX_POLYFILL:?}" in
+case "${BSONCXX_POLYFILL:-}" in
 mnmlstc) cmake_flags+=(-D "BSONCXX_POLY_USE_MNMLSTC=ON") ;;
 boost) cmake_flags+=(-D "BSONCXX_POLY_USE_BOOST=ON") ;;
 impls) cmake_flags+=(-D "BSONCXX_POLY_USE_IMPLS=ON") ;;
 std) cmake_flags+=(-D "BSONCXX_POLY_USE_STD=ON") ;;
 esac
 
-echo "Configuring with CMake flags: ${cmake_flags[*]}"
+echo "Configuring with CMake flags:"
+for flag in "${cmake_flags[@]}"; do
+  echo " - ${flag:?}"
+done
+echo
```

Comment:
Minor shorthand with `printf`:

```suggestion
echo "Configuring with CMake flags:"
printf " - %s\n" "${cmake_flags[@]}"
echo
```